R/api_client.R

```diff
@@ -140,6 +141,32 @@ DatabricksClient <- function(profile = NULL, host = NULL, token = NULL, config_f
     return(function() {
       c(Authentication = paste("Bearer", cfg$token))
     })
+  }, `databricks-cli` = function() {
+    token_source <- .databricks_cli_token_source(cfg)
+    if (is.null(token_source)) {
+      return(NULL)
+    }
+    result <- try(token_source$token(), silent = TRUE)
+    if (inherits(result, "try-error")) {
+      return(NULL)
+    }
+    return(function() {
+      token <- token_source$token()
+      return(token$headers())
+    })
+  }, `azure-cli` = function() {
+    if (!is_azure()) {
+      return(NULL)
+    }
+    token_source <- .azure_cli_token_source("2ff814a6-3304-4ab8-85cb-cd0e6f879c1d")
```

Comment:
should this be removed?

Reply:
Nope. It holds private state used for token refreshing internally when it's about to expire